bigtable/tests/mutation_integration_test.cc

```diff
@@ -0,0 +1,508 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/testing/table_integration_test.h"
+
+namespace {
+
+namespace admin_proto = ::google::bigtable::admin::v2;
+
+class MutationIntegrationTest : public bigtable::testing::TableIntegrationTest {
+ protected:
+  std::string column_family1 = "fam1";
+  std::string column_family2 = "fam2";
+  std::string column_family3 = "fam3";
+
+  // Create Table
+  bigtable::TableConfig table_config = bigtable::TableConfig(
+      {{column_family1, bigtable::GcRule::MaxNumVersions(10)},
+       {column_family2, bigtable::GcRule::MaxNumVersions(10)},
+       {column_family3, bigtable::GcRule::MaxNumVersions(10)}},
+      {});
+
+  /// A helper function to create a list of cells.
+  void CreateCellsWithoutTimestamp(bigtable::Table& table,
+                                   std::vector<bigtable::Cell> const& cells) {
+    std::map<std::string, bigtable::SingleRowMutation> mutations;
+    for (auto const& cell : cells) {
+      std::string key = cell.row_key();
+      auto inserted = mutations.emplace(key, bigtable::SingleRowMutation(key));
+      inserted.first->second.emplace_back(bigtable::SetCell(
+          cell.family_name(), cell.column_qualifier(), cell.value()));
+    }
+
+    bigtable::BulkMutation bulk;
+    for (auto& kv : mutations) {
+      bulk.emplace_back(std::move(kv.second));
+    }
+    table.BulkApply(std::move(bulk));
+  }
+
+  /*
+   * Delete the records from table for the specified family and column.
+   */
+  void DeleteFromColumn(bigtable::Table& table, bigtable::Cell cell) {
+    bigtable::SingleRowMutation mutation(cell.row_key());
+    mutation.emplace_back(bigtable::DeleteFromColumn(cell.family_name(),
+                                                     cell.column_qualifier()));
+
+    bigtable::BulkMutation bulk;
+    bulk.emplace_back(std::move(mutation));
+    table.BulkApply(std::move(bulk));
+  }
+
+  /*
```

Comment:
If this is supposed to be a Doxygen comment it is missing the `/**` at the beginning.